### DISCUSSION TITLE
Drag & drop file into webview

### DISCUSSION THREAD

**Discussion by EmberLightVFX** [zauberzeug/nicegui#3372] [Ideas / Feature Requests]
I really need to be able to drag and drop files and folders into the webview and get the objects full path.
It's impossible to get the objects path using javascript. It's a safety mechanism.

But it is possible to get the path using pywebview's dom https://pywebview.flowrl.com/examples/drag_drop.html !

I have tried adding the above code into `native_module.py` and it works as wanted!

The problem now is how to make the pywebview instance and NiceGUI communicate with each other.
app.native.main_window.dom always returns None for me...

The best would be if there is some way to fire events to get the data, like `ui.on("drop", my_func)`.

Thoughts?

**Comment by rodja** [comment]
Hello @EmberLightVFX. That sounds interesting. The communication between pywebview and NiceGUI is tricky because they must run in separate processes. We use `multiprocessing.Queue` for this: https://github.com/zauberzeug/nicegui/blob/874b5abfbbf1c7b6a4907c12d5d0389da74aaaec/nicegui/server.py#L15-L16

But firing an event from JavaScript may be a smart work around. @polaar had some experiment going in this regard: https://github.com/zauberzeug/nicegui/discussions/283#discussioncomment-5862401.

**Reply by EmberLightVFX** [reply]
I created a working PR here: https://github.com/zauberzeug/nicegui/pull/3378
I got the information out from the mutliprocessing thread but I'm stuck trying to get it into the current running ui-instance so I could run something like `ui.notify(item)`
